pulp_smash/tests/pulp3/file/api_v3/test_auto_distribution.py

```diff
@@ -0,0 +1,137 @@
+# coding=utf-8
+"""Tests related to auto distributions."""
+import hashlib
+import unittest
+from urllib.parse import urljoin
+
+from pulp_smash import api, config, utils
+from pulp_smash.constants import FILE_FEED_URL, FILE_URL
+from pulp_smash.tests.pulp3.constants import (
+    DISTRIBUTION_PATH,
+    FILE_CONTENT_PATH,
+    FILE_PUBLISHER_PATH,
+    FILE_REMOTE_PATH,
+    REPO_PATH,
+)
+from pulp_smash.tests.pulp3.file.api_v3.utils import gen_publisher, gen_remote
+from pulp_smash.tests.pulp3.file.utils import set_up_module as setUpModule  # pylint:disable=unused-import
+from pulp_smash.tests.pulp3.pulpcore.utils import gen_distribution, gen_repo
+from pulp_smash.tests.pulp3.utils import (
+    get_added_content,
+    get_auth,
+    get_versions,
+    publish,
+    sync,
+)
+
+
+class AutoDistributionTestCase(unittest.TestCase):
+    """Test auto distribution."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables.
+
+        Add content to Pulp.
+        """
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        remote = {}
+        repo = {}
+        try:
+            body = gen_remote(urljoin(FILE_FEED_URL, 'PULP_MANIFEST'))
+            remote.update(cls.client.post(FILE_REMOTE_PATH, body))
+            repo.update(cls.client.post(REPO_PATH, gen_repo()))
+            sync(cls.cfg, remote, repo)
+            cls.content = cls.client.get(FILE_CONTENT_PATH)['results']
```

Comment:
This doesn't need to be part of the try/finally block. In fact, it can be moved down into the `test*` method.